### PR TITLE
Add Haskell support

### PR DESCRIPTION
**Pull request by AlexBethel**
Adds support for executing Haskell snippets.

### CODE REVIEW

File: discord/dockerfiles/hs-Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM haskell:9-buster
```
Comment (qbxt) [inline discord/dockerfiles/hs-Dockerfile:1]: Is there an advantage to using `buster` over `slim-buster`?

File: discord/help.py [hunk 2]
```diff
@@ -13,10 +13,11 @@ def __init__(self, bot):
         self.bash = "<:bash:831116845133332500>"
         self.js = "<:nodejs:834358450309300265>"
         self.cpp = "<:cpp:917582153544507442>"
+        self.haskell = "<:haskell:310831928360304653>"
```
Comment (qbxt) [inline discord/help.py:16]: Will you please send me this emoji so I can put it in my private server with the other ones?

File: discord/dockerfiles/hs-Dockerfile (hunk 1 above)
Comment (AlexBethel) [reply discord/dockerfiles/hs-Dockerfile:1]: No, that was my mistake; for this use case it should probably be `9-slim-buster`. I'll go ahead and switch it over.

File: discord/help.py (hunk 2 above)
Comment (AlexBethel) [reply discord/help.py:16]: Yeah, I'll send it to you on Discord.

**Comment by qbxt** [comment]
I just added in the Java PR, would you please update your version to include the latest Java changes?

Also, please use the Discord emoji `<:haskell:955167472850464768>`

**Comment by AlexBethel** [comment]
All right, think I got the merge done correctly, and everything seems to still work as far as I can tell.

**Comment by qbxt** [comment]
LGTM, thanks for the PR

**Pull request merged**